eeco/utils.py

```diff
@@ -312,21 +310,238 @@ def const_rule(model):
+    model : pyomo.environ.Model
+        The Pyomo model containing the variables to initialize.
+
+    charge_dict : dict
+        Dictionary containing charge arrays for different utilities and charge types.
+        Used to extract the correct charge rate for export calculations.
+
+    Returns
+    -------
+    dict
+        Dictionary with initialized consumption objects for each utility.
+    """
+    consumption_object_dict = {}
+
+    # Initialize the basic consumption variables and converted variables
+    for utility in consumption_data_dict.keys():
+        consumption_data = consumption_data_dict[utility]
+        consumption_var = model.find_component(f"{utility}_consumption")
+        if consumption_var is not None:
+            for t in model.t:
+                consumption_var[t].value = consumption_data[t - 1]  # Pyomo 1-indexed
+        converted_var = model.find_component(f"{utility}_converted")
+        if converted_var is not None:
+            for t in model.t:
+                converted_var[t].value = consumption_data[t - 1]
+
+        consumption_object_dict[utility] = {}
+        consumption_data = consumption_data_dict[utility]
+
+        # Decompose using numpy version
+        positive_values, negative_values, _ = decompose_consumption(consumption_data)
+
+        # Find and initialize the corresponding Pyomo variables
+        pos_name, neg_name = _get_decomposed_var_names(utility)
+        positive_var = model.find_component(pos_name)
+        negative_var = model.find_component(neg_name)
+
+        for t in model.t:
+            positive_var[t].value = positive_values[t - 1]
+            negative_var[t].value = negative_values[t - 1]
+
+        consumption_object_dict[utility]["imports"] = positive_var
+        consumption_object_dict[utility]["exports"] = negative_var
+
+    # # Initialize export-related variables created by calculate_export_revenue
+    for component_name in model.component_map():
+        if "_multiply" in component_name:
+            component = model.find_component(component_name)
+            if hasattr(component, "__iter__") and hasattr(
+                component[list(component.keys())[0]], "value"
+            ):
+                for i in component:
+                    export_var = model.find_component("electric_negative")
+                    if export_var is not None and hasattr(export_var[i], "value"):
+                        charge_rate = 0.0  # Default export
+                        if charge_dict is not None:
+                            export_keys = [
+                                key for key in charge_dict.keys() if "export" in key
+                            ]
+                            if export_keys:
+                                charge_rate = charge_dict[export_keys[0]][0]
+                        component[i].value = export_var[i].value * charge_rate
+                    else:
+                        component[i].value = 0.0
+        if "_sum" in component_name:
```

Comment:
Can you also as to what `_sum` and `_multiply` are signifying about the component? I don't understand why we are replacing `_sum` with `_multiply`